src/main/java/core/basesyntax/service/operation/PurchaseOperation.java

```
@@ -0,0 +1,12 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.db.Storage;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(String fruitName, int amount) {
+        int currentAmount = Storage.get(fruitName);
+        int newAmount = currentAmount - amount;
```

Consider adding a check to ensure that `newAmount` is not negative. If `amount` exceeds `currentAmount`, it should throw an exception or handle the error appropriately to prevent negative stock values.